shared/src/main/scala/coop/rchain/shared/StreamT.scala

```diff
@@ -29,6 +29,19 @@ sealed abstract class StreamT[F[_], +A] { self =>
     case _: SNil[F]      => StreamT.empty[F, A]
   }
 
+  def filterF[AA >: A](p: A => F[Boolean])(implicit monad: Monad[F]): F[StreamT[F, AA]] =
+    self match {
+      case SCons(curr, lazyTail) =>
+        p(curr).flatMap { cond =>
+          if (cond)
+            monad.pure[StreamT[F, AA]](StreamT.cons(curr, lazyTail.map(_.flatMap(_.filterF(p)))))
+          else
+            lazyTail.value.flatMap(_.filterF(p))
```

Comment:
I don't think `StreamT` should eagerly try to find the first element after the filter is applied. It think it would be better to `map` over the tail instead of forcing its evaluation with `.value`.